src/lib/datatypes/sol-buffer.c

```diff
@@ -342,3 +342,33 @@ sol_buffer_ensure_nul_byte(struct sol_buffer *buf)
 
     return 0;
 }
+
+SOL_API int
+sol_buffer_append_char(struct sol_buffer *buf, const char c)
+{
+    const size_t nul_size = nul_byte_size(buf);
+    char *p;
+    size_t new_size;
+    int err;
+
+    SOL_NULL_CHECK(buf, -EINVAL);
+
+    err = sol_util_size_add(buf->used, 1, &new_size);
+    if (err < 0)
+        return err;
+
+    /* Extra room for the ending NUL-byte. */
```

Comment:
Ederson added `sol_buffer_ensure_nul_byte()` to do this part. Only append `c` and call that function.

Also, take the time to add the `insert_char` version.